lightning/src/onion_message/packet.rs

```diff
@@ -136,6 +136,19 @@ impl<T: CustomOnionMessageContents> OnionMessageContents<T> {
 	}
 }
 
+impl<T: CustomOnionMessageContents> fmt::Display for OnionMessageContents<T> {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            OnionMessageContents::Offers(ref msg) => {
+                write!(f, " (Offers): {:?}", msg)
+            }
+            OnionMessageContents::Custom(ref msg) => {
+                write!(f, " (Custom) : {}", msg.tlv_type())
```

Comment:
Could we improve on this?  Currently, in one of the tests I see:

```
TRACE node 0 [lightning::onion_message::messenger : lightning/src/onion_message/messenger.rs, 347] Sending onion message:  (Custom) : 4343
TRACE node 1 [lightning::onion_message::messenger : lightning/src/onion_message/messenger.rs, 495] Received an onion message  (Custom) : 4343 with path_id None and no reply_path
```

But it's unclear what `4343` means.

BTW, I'd recommend formatting with `Formatter::debug_struct` and related methods for more canonical formatting whenever it makes sense.

https://doc.rust-lang.org/core/fmt/struct.Formatter.html#method.debug_struct

Might be obsoleted by another comment, but in general we should use this.

Reply:
Here `4343` can be `const CUSTOM_RESPONSE_MESSAGE_TYPE: u64 = 4343;`

Reply:
That's just used in the test. In practice, users will parameterize their own custom message handler which can use any value.

Reply:
Ok , so anything need to be done here? Or i update the PR .